cmd/fake/main.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var counts, errorRate, errors int
+var errorCodes []int
+
+func replyOK(w http.ResponseWriter, r *http.Request) {
+	message := strings.TrimPrefix("/", r.URL.Path)
+	message = message + " served ok"
+
+	responseCode := http.StatusOK
+	w.WriteHeader(responseCode)
+	w.Write([]byte(message))
+}
+
+func replyError(w http.ResponseWriter, r *http.Request) {
+	message := strings.TrimPrefix("/", r.URL.Path)
+	message = "Errored request to " + message
+
+	responseCode := errorCodes[rand.Intn(len(errorCodes))]
```

Comment:
error codes provided. It would also allow some control over the
order of codes to be presented.